src/worker.js

```diff
@@ -136,15 +136,17 @@ this.onmessage = function(e) {
       }
 #endif
 
-#endif
+#endif // WASM
 
 #if !MINIMAL_RUNTIME || MODULARIZE
       {{{ makeAsmImportsAccessInPthread('ENVIRONMENT_IS_PTHREAD') }}} = true;
 #endif
 
 #if MODULARIZE && EXPORT_ES6
       import(e.data.urlOrBlob).then(function({{{ EXPORT_NAME }}}) {
-        Module = {{{ EXPORT_NAME }}}.default(Module);
+        return {{{ EXPORT_NAME }}}.default(Module);
+      }).then(function(instance) {
+        Module = instance;
```

Comment:
why this change?

Reply:
Before, calling the factory function that is the default export synchronously returned the instance.  Now it returns a promise. We return it from the promise handler instead of calling `then` on it directly in order to not nest the code any further.